.github/workflows/ci.yml

```diff
@@ -43,10 +43,16 @@ jobs:
           # https://github.com/actions/runner/issues/409#issuecomment-752775072
           components: ${{ contains(matrix.channel, 'nightly') && 'miri' || '' }}
 
+    - name: Set feature string for cache key
+      run: |
+        features=${{matrix.features}}
+        features_key=${#features}
+        echo "FEATURES=$(echo $features_key)" >> $GITHUB_ENV
```

Comment:
This is really clever! So long as you're executing arbitrary shell commands, maybe you could just hash the feature set instead so this will keep working even if we have two feature sets of the same character length?

```yaml
    - name: Set feature string for cache key
      run: |
        echo "FEATURES_HASH=$(echo ${{ matrix.features }} | sha256sum | cut -d ' ' -f 1)" >> $GITHUB_ENV
```

Reply:
That's actually a great idea, just added that!